### ISSUE TITLE
LR scheduler clarification

### ISSUE THREAD

**Issue by eldarkurtic** [IntelLabs/Model-Compression-Research-Package#5]
Hi,
Running the Language Modelling example (https://github.com/IntelLabs/Model-Compression-Research-Package/tree/main/examples/transformers/language-modeling) ends with a slightly different LR schedule compared to the one presented in the Figure 2.b of the "Prune Once For All" paper. (particularly the warmup phase seems to be a bit different)

`train/learning_rate` logged by Weights&Biases:
<img width="652" alt="Screenshot 2021-12-20 at 11 25 39" src="https://user-images.githubusercontent.com/8884008/146752997-b7ab76d9-d5d4-42d7-9773-97e050400a1d.png">

Learning rate in the paper, Figure 2.b:
<img width="490" alt="Screenshot 2021-12-20 at 11 31 35" src="https://user-images.githubusercontent.com/8884008/146753341-31280a47-d175-4544-aaeb-2ac23d514d81.png">

**Comment by ofirzaf** [comment]
Hi,

The figure in the paper was only an example and it was not generated from either of the experiments presented in the paper. The learning rate rewinding mechanism kicks in at the begining of the pruning policy and stops at the end of your pruning policy. Pruning policy explained in #4.

So, I assume from the figure you posted that your pruning policy begins at the same time you start your training and ends at around `global_step=47k`. At this timestep you will rewind your learning rate according to the defined `pruning_frequency` (not a great name for that parameter) in the pruning config. The jigsaw like pattern is visible in your figure so I guess everything works in order. Depending on the sampling interval of your logger you might miss the pattern. For example, if you rewind every 1k steps and log every 1k steps you will get a straight line.

**Comment by eldarkurtic** [comment]
So the learning rate rewinding frequency is equal to the `pruning_frequency` right? 
Something like: start with LR=1.5e-4 and linearly decay for `pruning_frequency` number of steps, then prune the model, and rewind the learning rate back to 1.5e-4?

One more question regarding this: what would be the min LR value reached before the rewind is triggered?

**Comment by ofirzaf** [comment]
Yes, that is correct.
I have never checked what is the min value the LR reached.
